platform/mellanox/hw-management/Makefile

```diff
@@ -6,7 +6,8 @@ MAIN_TARGET = hw-management_1.mlnx.$(MLNX_HW_MANAGEMENT_VERSION)_amd64.deb
 $(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
 
 	pushd hw-mgmt
-	git am ../*.patch
+	git stash
```

Comment:
What is the purpose of `git stash`? Do you intend to `git reset --hard`?

Reply:
The idea was that if I develop a patch for hw-mgmt and have some changes that are not commited the stash will save them.

Reply:
This will impact the whole local repo. Assume other developers modify some file in this repo locally, and build locally, if hw-managment is also built in the process, the local changes will be stashed, and it is surprising.

Reply:
I notice the same issue on master branch. So if you agree, please fix on master first.

Reply:
@qiluo-msft  If we look at other 3rd party projects built with patches in SONiC, e.g libteam the whole libteam is deleted every time you build which might also be a surprise if modifying libteam locally. ```git reset --hard``` would be even more destructive because local changes are not saved.

Reply:
Sorry I just notice `git stash` in submodule, not in the sonic-buildimage repo. Then it will mostly not impact other developers.